index.rst

```diff
@@ -1,61 +1,383 @@
+
+For options on how to implement this, and some further discussion, see `SQR-051`_.
+The recommended approach is to filter out incoming cookies by replacing the ``Cookie`` header with a rewritten header and replacing or dropping the ``Authorization`` header.
+
+.. _SQR-051: https://sqr-051.lsst.io/
+
+Logging
+-------
+
+Because authentication and access control is done via a generic layer in front of all applications, it can log all authenticated operations and maintain data about how tokens are used.
+This, in turn, will be used to investigate possible security incidents and to look for anomolies in how tokens are used (such as use from an unexpected IP address).
+
+Isolation
+=========
+
+Web security is closely tied to origins, which are (roughly) defined as the tuple of protocol, hostname, and port.
+All JavaScript running on the same origin (including any URL path in that origin) has effectively the same privileges, can freely make HTTP requests of any type to other URLs on the same origin, can access any local state tied to that origin, and so forth.
+
+Cookies have a similar concept called a site, which controls when the browser will send the cookie alongside a request.
+The site of a cookie is somewhat less restricted than a JavaScript origin and can include subdomains depending on the cookie configuration.
+
+The effect of this security model is that it is not possible to secure web applications against malicious JavaScript running in the same origin.
+To isolate one web application from another, they must run in separate origins.
+
+The easiest way to do this is to give every application a separate origin, usually by changing the hostname.
+In the case of the Science Platform, this would mean assigning a separate hostname to every application, and then using either multiple TLS certificates, a wildcard TLS certificate, or a TLS certificate with multiple :abbr:`SANs (Subject Alternative Names)`.
+
+However, separate origins are only crucial for web applications that run JavaScript in the browser.
+REST API endpoints can safely share the same origin provided that they do not need to support cross-origin requests, do not serve pages or JavaScript that run in a web browser, and do not share an origin with a web application that does so.
+Having all Science Platform REST APIs share the same origin is useful for documentation purposes and allows more flexibility about which API endpoints are served by which backend implementations.
+
+Therefore, the isolation plan for the Rubin Science Platform is:
+
+- Serve the Notebook Aspect spawning interface from its own origin
+- Serve each user's notebook from a per-user origin (see :ref:`jupyterlab-origin`)
+- Serve the Portal Aspect from its own origin
+- Serve the authentication system from its own origin
+- Serve all APIs from a single origin shared by the APIs, but separate from the other origins
+
```

Comment:
I don't know if it's worth explicitly discussing this, but what if we had a saphir/roundtable service for microservices or 3rd party-to-RSP but codeployed apps (like the alert filtering service)? Not sure whether we should demand for all of them to be in separate origins, or an origin for all these "guest" services (and also unclear as to whether this would make the safir/templating layer's job harder or not"

Reply:
They can all share an origin with trusted APIs as well, as long as nothing on that origin serves any JavaScript. I'll add another paragraph explaining this. The only thing we need to separate into its own origins is services that serve a browser UI.